From ca6176873b9105f31d4b56f76db81ada770a0162 Mon Sep 17 00:00:00 2001
From: Rob Vogelaar <revogelaar@gmail.com>
Date: Wed, 2 Jul 2025 14:51:56 -0700
Subject: [PATCH] platform_hal: Switch from /proc/cpuinfo to /etc/vcpe-config
 for device info

- Change GetSerialNumber to read SERIAL_NUMBER from /etc/vcpe-config
- Change GetHardwareVersion to read HARDWARE_VERSION from /etc/vcpe-config
- Replace execute() calls with execute_cmd() for consistency

Signed-off-by: Rob Vogelaar <revogelaar@gmail.com>
---
 source/platform/platform_hal.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/source/platform/platform_hal.c b/source/platform/platform_hal.c
index 8ca224e..9aa0ba1 100644
--- a/source/platform/platform_hal.c
+++ b/source/platform/platform_hal.c
@@ -220,7 +220,7 @@ INT platform_hal_GetSerialNumber(CHAR* pValue)
     {
         return RETURN_ERR;
     }
-    ret = execute("grep 'Serial' /proc/cpuinfo", sn);
+    ret = execute_cmd("grep 'SERIAL_NUMBER' /etc/vcpe-config | cut -d '=' -f2", sn);
     if(RETURN_OK != ret)
     {
         printf("\nError %s\n", __func__);
@@ -242,7 +242,7 @@ INT platform_hal_GetHardwareVersion(CHAR* pValue)
         return RETURN_ERR;
     }
 
-    ret = execute("grep 'Revision' /proc/cpuinfo", hwVer);
+    ret = execute_cmd("grep 'HARDWARE_VERSION' /etc/vcpe-config | cut -d '=' -f2", hwVer);
     if(RETURN_OK != ret)
     {
         printf("\nError %s\n", __func__);
-- 
2.25.1

